engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/UnresolvedConstructor.java

```diff
@@ -158,4 +162,15 @@ Object instantiateUncached(
       return r;
     }
   }
+
+  @ExportMessage
+  boolean hasType() {
+    return true;
+  }
+
+  @ExportMessage
+  Type getType(@Bind("$node") Node node) {
```

Comment:
Per suggestion of Christian Humber:

> you should use @Bind("$node") for context references in libraries instead of arbitrary nodes. In general they might not be adoptable:

```java
  @ExportLibrary(value = TypesLibrary.class, receiverType = Long.class)
    public static class DefaultLongExports {

        @ExportMessage
        static boolean hasType(Long receiver) {
            return true;
        }

        @ExportMessage
        static Class<?> getType(Long receiver, @Bind("$node") Node node) {
            SLContext.get(node);
            return Long.class;
        }
    }
```
we don't have to use [@Cached("1") int ignore](https://github.com/enso-org/enso/blob/c4029eed4f07318f5139e389e2554b95b85ccfc3/engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Ref.java#L66) anymore...

Reply:
... if it works. Sometimes [it doesn't work](https://github.com/enso-org/enso/pull/9431/files#diff-6236fe324f03fbe4b5a54ae6264e55583c3b3e3b9f2d58e84d302789c451a89fR135) and I had to add the `@Cached("1") int ignore` trick back.